### PR TITLE
Enabled extended CI

### PR DESCRIPTION
**Pull request by artemry-nv**
Signed-off-by: artemry-nv <artemry@nvidia.com>

## What
Enabled extended CI with UCC unit tests, Torch-UCC and DLRM tests.

## Why ?
Improve UCC quality with increased test coverage.

### CODE REVIEW

**Comment by artemry-nv** [comment]
@Sergei-Lebedev @vspetrov please review.

File: .ci/scripts/configure_python.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Install conda python
+#conda update -y conda
+#conda install -c anaconda -y \
+#    python \
+#    pip \
+#    scikit-learn
+#pip3 install --no-cache-dir python-hostlist
+
+#alternatives --set python /opt/conda/bin/python3
+alternatives --set python /usr/bin/python3
```
Comment (mike-dubman) [inline .ci/scripts/configure_python.sh:20]: IMO you need run 1st ``` alternatives --install /usr/bin/python python /usr/bin/python3 1```

Comment (artemry-nv) [reply .ci/scripts/configure_python.sh:20]: Could you please elaborate a bit why we need this additional command?
```
# alternatives --install /usr/bin/python python /usr/bin/python3 1
the primary link for python must be /usr/bin/unversioned-python

# update-alternatives --config python
There are 2 programs which provide 'python'.
  Selection    Command
-----------------------------------------------
*  1           /usr/libexec/no-python
 + 2           /usr/bin/python3
Enter to keep the current selection[+], or type selection number: ```

**Comment by artemry-nv** [comment]
@Sergei-Lebedev @vspetrov could you please merge this PR? Then I'll enable corresponding Jenkins job.

File: .ci/scripts/run_tests_ucc_mpi.sh [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+# shellcheck disable=SC2086
+mpirun \
+    -np $NP \
+    --hostfile ${HOSTFILE} \
+    --map-by node \
+    --allow-run-as-root \
+    --mca plm_rsh_args '-p 12345' \
+    -x PATH \
+    -x LD_LIBRARY_PATH \
+    /opt/nvidia/torch-ucc/src/ucc/build/test/mpi/ucc_test_mpi --mtypes host --root random:2 --count_bits 32,64 --displ_bits 32,64
```
Comment (vspetrov) [inline .ci/scripts/run_tests_ucc_mpi.sh:60]: lets add " -I 2" (capital "i") - this will run with and w/o INPLACE

**Review by vspetrov — APPROVED** [review]

File: .ci/scripts/run_param_benchmarks.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+exit 0
+
+source /workspace/set-env-dist.sh
+index=$LOCAL_RANK
+export OMPI_COMM_WORLD_SIZE=$WORLD_SIZE
+export OMPI_COMM_WORLD_LOCAL_SIZE=$LOCAL_SIZE
+export OMPI_COMM_WORLD_RANK=$RANK
+export OMPI_COMM_WORLD_LOCAL_RANK=$LOCAL_RANK
+
+if (( $index == 0 )); then
```
Comment (Sergei-Lebedev) [inline .ci/scripts/run_param_benchmarks.sh:14]: do we really have so many cores and HCAs on CI node?

**Review by Sergei-Lebedev — APPROVED** [review]

File: .ci/scripts/run_tests_ucc_mpi.sh (hunk 2 above)
Comment (artemry-nv) [reply .ci/scripts/run_tests_ucc_mpi.sh:60]: Fixed.

File: .ci/scripts/run_param_benchmarks.sh (hunk 3 above)
Comment (artemry-nv) [reply .ci/scripts/run_param_benchmarks.sh:14]: Removed this script (it's just a placeholder for now).

**Comment by artemry-nv** [comment]
Could anyone please merge this PR (need to proceed with further CI customization)?

**Comment by Sergei-Lebedev** [comment]
> Could anyone please merge this PR (need to proceed with further CI customization)?

please fix commit title, it didn't pass codestyle check.
I think following should work "TEST: enabled extended CI"

**Comment by artemry-nv** [comment]
> > Could anyone please merge this PR (need to proceed with further CI customization)?
> 
> please fix commit title, it didn't pass codestyle check.
> I think following should work "TEST: enabled extended CI"

Thanks, fixed.

**Pull request merged**